### ISSUE TITLE
Error on signTransaction from @ledgerhq/hw-app-eth for Binance Smart Chain

### ISSUE THREAD

**Issue by fabenedetti** [LedgerHQ/ledgerjs#806]
Hi, I'm trying to implement the signing of a transaction for Binance Smart Chain but I'm receiving an error.

I noticed that there is a Binance Smart Chain app so I installed the latest version and updated the Ethereum app (both are on v1.9.17) on my Ledger device. On both apps, I have **enabled Blind signing** on the settings.

However, whenever I try to sign a BSC transaction using the Binance Smart Chain app on the device, it fails and returns an error 
`Please enable Blind signing or Contract data in the Ethereum app Settings`. I saw the same behaviour when I tried to transfer funds with my Ledger (BSC app) using Metamask. Also, I noticed that they suggest to use Ethereum app.

![metamask_bsc](https://user-images.githubusercontent.com/7342345/156621475-4f9d1745-fb65-42a2-8f84-316fdd00b04a.png)
![metamask_fail](https://user-images.githubusercontent.com/7342345/156621494-e64dd67c-062e-49cb-a791-29261338929b.png)

If I try to sign the tx using the **Ethereum app** instead of the **Binance Smart Chain app** it works as expected, I tested sending transactions over testnet and they were confirmed.

I couldn't find much documentation about signing BSC txs, most of the docs are related to Ethereum.

So, I wanted to confirm with you if this is could be a bug or if I should use Ethereum app for signing txs on the BSC network. 

@gre

**Comment by fedecaccia** [comment]
Experimenting same issue here, @gre does ledgerHQ has a solution for this?

**Comment by gre** [comment]
on the device, if you go on the Settings of the app, did you enable blind signing? as far as I know, this is the device behaivor when you sign an "unrecognized" contract. https://support.ledger.com/hc/en-us/articles/4405481324433-Enable-blind-signing-in-the-Ethereum-ETH-app?docs=true

**Comment by fabenedetti** [comment]
@gre yes, I have enabled Blind signing on both apps on the device (Ethereum app and Binance Smart Chain app) following the the steps on that document. Whenever I try to sign, it works well with the Ethereum app but it fails with the Binance Smart Chain app. I even tried reinstalling the app and re-enabling that setting.

**Comment by fabenedetti** [comment]
@gre I have just realized that it only fails with a tx for BSC testnet. Perhaps, on the Binance Smart Chain app there is a bug or a strict validation for chainId 56 (mainnet BSC) and when I try to sign a tx with chainId 97 it isn't recognized and throws the mentioned error. This is working well on the Ethereum app (signing tx for BSC testnet). Could you confirm if this is an expected behaviour or if it's a bug ?
